Support TLS connections for Postgres sources

uce opened the pull request:
Opening this as a draft while I work on the `rust-postgres` client changes that block the remaining changes. @petrosagg Do the tests and the places where we create the TLS connector look sane?

## TODO
- [x] Use existing test certs infra (just saw this after opening the PR)
- [ ] Add remaining sslmodes to `rust-postgres`

petrosagg commented:
Yes, this looks great! One thing that comes to mind is that we don't need to duplicate the full testdrive tests for both the ssl and non-ssl version. We just need two small tests testing the connection modes and then run the full suite that verify the correctness of the data with SSL.

Also, what do you think of finishing and merging the support for the three modes already supported (Disable, Prefer, Require) before implementing the more complex ones? This will allow users actively testing this feature to use it with RDS databases

uce commented:
I updated the PR.

> Yes, this looks great! One thing that comes to mind is that we don't need to duplicate the full testdrive tests for both the ssl and non-ssl version. We just need two small tests testing the connection modes and then run the full suite that verify the correctness of the data with SSL.

As discussed offline, we kept the tests as they are and moved the existing tests to require SSL.

> Also, what do you think of finishing and merging the support for the three modes already supported (Disable, Prefer, Require) before implementing the more complex ones? 

I updated the PR. Unfortunately, I didn't get the test-certs to work directly (because of file perms), but didn't want to block the PR on it. I might give it another shot after this is merged.

petrosagg commented on src/postgres-util/src/lib.rs:
```diff
@@ -66,7 +77,9 @@ pub async fn publication_info(
     publication: &str,
 ) -> Result<Vec<TableInfo>, anyhow::Error> {
-    let (client, connection) = tokio_postgres::connect(&conn, NoTls).await?;
+    let config = conn.parse::<Config>()?;
```
Is this type annotation necessary?

uce replied:
That was a left-over. Thanks for catching 👍🏼

petrosagg commented on test/pg-cdc/mzcompose.yml:
```diff
@@ -32,13 +45,34 @@ services:
+    depends_on: [postgres-prepare-certs]
+
+  # Postgres runs as non-root. Is there a better way to fix these perms?
+  postgres-prepare-certs:
```
I'm not sure if it would make this setup simpler, but the postgres docker image supports adding initialization scripts in it that run automatically. `rust-postgres` uses this mechanism to set up certificates, `pg_hba.conf`, and users.

This is the script https://github.com/sfackler/rust-postgres/blob/master/docker/sql_setup.sh which is then imported here https://github.com/sfackler/rust-postgres/blob/master/docker/Dockerfile#L3

uce replied:
Nice! I hadn't seen this before. I think it will make things quite a bit simpler in combination with mzbuild (assuming I can depend on `test-certs` directly). Let me try that real quick and update if possible.

philip-stoev replied:
You can also check out ```test/kafka-ssl/mzcompose.yml``` where the testdrive container entrypoint is used to put certs into position.

uce replied:
As discussed offline, the approach from `test/kafka-ssl/mzcompose.yml` does not carry over, because we run testdrive after postgres has started.

It took a bit longer than expected, but I was able to simplify the `mzcompose.yml` by introducing a `postgres` mzimage.

petrosagg reviewed (approved):
This looks great! thank you

philip-stoev reviewed (approved):
Thank you for the tests!
